library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -337,7 +337,7 @@ protected final void maybeInitCodec() throws ExoPlaybackException {
     }
 
     String codecName = decoderInfo.name;
-    codecIsAdaptive = decoderInfo.adaptive;
+    codecIsAdaptive = decoderInfo.adaptive && codecSupportsAdaptive(codecName, format);
```

Comment:
&& !codecNeedsDisableAdaptationWorkaround(codecName)

codecSupportsAdaptive is misleading as a method name because it returns true for pretty much everything, including all non-adaptive decoders :).